pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,336 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/go-systemd/v22/unit"
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/service"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+type K8sPlugin struct {
+	PluginName  string
+	SpecVersion string
+}
+
+type McoService struct {
+	Name     string
+	Contents string
+}
+
+type McoFile struct {
+	Path     string
+	Contents struct {
+		Inline string
+	}
+}
+
+type McoServiceInjection struct {
+	Name    string
+	Dropins []struct {
+		Contents string
+	}
+}
+
+type k8sUpdateTarget struct {
+	switchdevService bool
+	switchdevScript  bool
+	networkManager   bool
+	openVSwitch      bool
+}
+
+func (u *k8sUpdateTarget) needUpdate() bool {
+	return u.switchdevService || u.switchdevScript || u.networkManager || u.openVSwitch
+}
+
+const (
+	mcoManifestPath           = "bindata/manifests/machine-config/"
```

Comment:
let rename the folder to be bindata/manifests/switchdev-config/
and MCO andK8s both of them will use it.